lib/src/Uncertainty/Algorithm/Simulation/ProbabilitySimulationAlgorithm.cxx

```diff
@@ -89,6 +99,13 @@ String ProbabilitySimulationAlgorithm::__repr__() const
 /* Compute the block sample and the points that realized the event */
 Sample ProbabilitySimulationAlgorithm::computeBlockSample()
 {
+  if (isExperimentProvided_)
+    return computeBlockSampleComposite();
+  return event_.getSample(blockSize_);
+}
+
+Sample ProbabilitySimulationAlgorithm::computeBlockSampleComposite()
+{
   Point weights;
   const Sample inputSample(experiment_.generateWithWeights(weights));
```

Comment:
Maybe a (parallel) contains() method acting on Sample could be added to the Domain class? It can speed-up a lot the case where the domain is a complex mesh.